### PR TITLE
pkg/trace: fix TruncateResource

### PR DESCRIPTION
**Pull request by bmermet**
### What does this PR do?

Fix a bug that inversed the logic of the ok return in `TruncateResource`

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### CODE REVIEW

File: pkg/trace/traceutil/truncate_test.go [hunk 1]
```diff
@@ -20,3 +20,20 @@ func TestTruncateString(t *testing.T) {
 	assert.Equal(t, "ééééé", TruncateUTF8("ééééé", 10))
 	assert.Equal(t, "ééé", TruncateUTF8("ééééé", 6))
 }
+
+func TestTruncateResource(t *testing.T) {
+	t.Run("below_limit", func(t *testing.T) {
```
Comment (gbbr) [inline pkg/trace/traceutil/truncate_test.go:25]:
```suggestion
	t.Run("over", func(t *testing.T) {
```

File: pkg/trace/traceutil/truncate_test.go [hunk 2]
```diff
@@ -20,3 +20,20 @@ func TestTruncateString(t *testing.T) {
...
+		r, ok := TruncateResource("resource")
+		assert.True(t, ok)
+		assert.Equal(t, "resource", r)
+	})
+	t.Run("above_limit", func(t *testing.T) {
```
Comment (gbbr) [inline pkg/trace/traceutil/truncate_test.go:30]:
```suggestion
	t.Run("under", func(t *testing.T) {
```

File: pkg/trace/traceutil/truncate_test.go [hunk 3]
```diff
@@ -20,3 +20,20 @@ func TestTruncateString(t *testing.T) {
 	assert.Equal(t, "ééééé", TruncateUTF8("ééééé", 10))
 	assert.Equal(t, "ééé", TruncateUTF8("ééééé", 6))
 }
+
+func TestTruncateResource(t *testing.T) {
+	t.Run("below_limit", func(t *testing.T) {
+		r, ok := TruncateResource("resource")
+		assert.True(t, ok)
+		assert.Equal(t, "resource", r)
+	})
```
Comment (gbbr) [inline pkg/trace/traceutil/truncate_test.go:29]: Empty line between sub-test please

File: pkg/trace/traceutil/truncate_test.go [hunk 4]
```diff
@@ -20,3 +20,20 @@ func TestTruncateString(t *testing.T) {
...
+	t.Run("above_limit", func(t *testing.T) {
+		s := ""
+		for i := 0; i < MaxResourceLen; i++ {
+			s += "a"
+		}
```
Comment (gbbr) [inline pkg/trace/traceutil/truncate_test.go:34]:
```suggestion
		s := strings.Repeat("a", MaxResourceLen)
```

**Review by gbbr — CHANGES_REQUESTED** [review]

File: pkg/trace/traceutil/truncate_test.go (hunk 4 above)
Comment (bmermet) [reply pkg/trace/traceutil/truncate_test.go:34]: Nice, I didn't know that one.

**Review by gbbr — APPROVED** [review]

**Pull request merged**